github-actions/unified-status-check/action.yml

```diff
@@ -0,0 +1,11 @@
+name: 'Unified Status Check'
+description: 'Monitor the statuses on a pull request to ensure all expected stauses are passing'
+author: 'Angular'
+inputs:
+  angular-robot-key:
```

Comment:
should the action not take a list of statuses that should be enforced? this would be similar to the existing robot and I think this is the key feature.

(I know at the same time GitHub branch protections could be used too- even for the current code in this PR? do you recall what the value of the status was?)

Reply:
I think we should flip the model and have it be an ignore list rather than a include list.

Reply:
I don't know I'd agree with it. I think we should clarify why we have the action vs. branch protection required status checks.

Even with a list of "required checks", all other non-required could be incorporated. I think that is how it's currently working in the gh robot.

Reply:
> My understanding/intention with this is that we use this to enforce "expected" statuses, but allowing for us to build out the contextual awareness of the check. For instance, allowing for ignoring when the pull request is in a draft state.

That's interesting. At the current time our contextual awareness would only be `if (isDraft) ? A : B`. With the current configuration model I cannot imagine more conditional enforcements (that can change) -- I see that argument 👍 

> As for the ignore rather than include list, I think its a better model to always include the checks and intentionally opt them out rather than creating a brittle set up of a name change in one of the statuses making it impossible for it to pass.

I would disagree with this. A good example is if e.g. CircleCI is not triggered at all, then the check would not see any failure/pending statuses and would show as "green".

Reply:
I suppose thats true, I wonder if there is a way to meet in the middle of it so that we don't create a brittle system but we make sure that we don't have false successes from just nothing running.

Let me see what I can do on that.

Reply:
Added the ability to do both ignore or match for required